modules/caddyhttp/rewrite/caddyfile.go

```diff
@@ -25,20 +27,116 @@ func init() {
 
 // parseCaddyfile sets up the handler from Caddyfile tokens. Syntax:
 //
-//     rewrite [<matcher>] <to>
+//     rewrite [<matcher>] [<to>] {
+//         to                <string>
+//         method            <string>
+//         strip_path_prefix <string>
+//         strip_path_suffix <string>
+//         rehandle          <bool>
+//         http_redirect     <status_code>
+//     }
 //
-// The <to> parameter becomes the new URI.
 func parseCaddyfile(h httpcaddyfile.Helper) (caddyhttp.MiddlewareHandler, error) {
-	var rewr Rewrite
+	var (
+		rewr Rewrite
+
+		toIsSet        bool
+		nextBlockIsSet bool
+		rehandleIsSet  bool
+		redirectIsSet  bool
+	)
+
 	for h.Next() {
-		if !h.NextArg() {
+		args := h.RemainingArgs()
+		switch len(args) {
+		case 0:
+		case 1:
+			toIsSet = true
+			rewr.URI = h.Val()
+		default:
 			return nil, h.ArgErr()
 		}
-		rewr.URI = h.Val()
-		if h.NextArg() {
-			return nil, h.ArgErr()
+
+		for h.NextBlock(0) {
+			nextBlockIsSet = true
+			switch h.Val() {
+			case "to":
+				if toIsSet {
+					return nil, h.Err("to is already set")
+				}
+				if !h.Args(&rewr.URI) {
+					return nil, h.ArgErr()
+				}
+			case "method":
+				if !h.Args(&rewr.Method) {
+					return nil, h.ArgErr()
+				}
+				switch rewr.Method {
+				case "GET", "HEAD", "POST", "PUT", "DELETE",
+					"CONNECT", "OPTIONS", "TRACE", "PATCH":
+				default:
+					return nil, h.Errf("unknown method '%s'", rewr.Method)
+				}
+			case "strip_path_prefix":
+				if !h.Args(&rewr.StripPathPrefix) {
+					return nil, h.ArgErr()
+				}
+			case "strip_path_suffix":
+				if !h.Args(&rewr.StripPathSuffix) {
+					return nil, h.ArgErr()
+				}
+			case "rehandle":
+				rehandleIsSet = true
+				if redirectIsSet {
+					return nil, h.Err("http_redirect is already set")
+				}
+
+				var rehandle string
+				if !h.Args(&rehandle) {
+					return nil, h.ArgErr()
+				}
+
+				switch rehandle {
+				case "true":
+					rewr.Rehandle = true
+				case "false":
+				default:
+					return nil, h.Errf("invalid rehandle argument '%s', expected bool", rehandle)
```

Comment:
For booleans in the Caddyfile, I like to simply use the presence of the subdirective to indicate a true value.

Reply:
I actually wonder if we need to / should expose this in the Caddyfile at all, since rehandling is kind of a detail of the underlying JSON routes, which is abstracted away from the user...

Reply:
That's an easy enough fix to check for presence.

In my case I need to prevent it from re-handling rewrites so an upstream system can be served under a path prefix, and strip the prefix from requests before sending them on their way. Unless there's a more appropriate way to handle that.

Reply:
Ah. See #2911 -- I think that we'll turn off rehandling by default and make it opt-in...

Reply:
I'll change it to check for presence only. I think the `route` directive would fit my use case.